pkg/client/fake/client.go

```diff
@@ -29,8 +29,7 @@ import (
 	"sync"
 	"time"
 
-	// Using v4 to match upstream
```

Comment:
/hold

This was intentional, see the godoc comment. Did upstream move to v5?

Reply:
I saw it, but I concluded that it was stale comment because in go.mod there was only v5/v5+incompatible. Now that I know there was some issue with newer versions of this lib (thanks @inteon) I'll research what to do next.

Reply:
The reason we use the older version here was that we copied code from upstream and wanted it to use the same library as upstream. 

I wasn't aware of general issues in the library. I wonder if we should re-visit our usage of v5 with this in mind.

Reply:
cc @vincepri @alvaroaleman We should probably figure out if we have to do something before we cut v0.17

Reply:
https://github.com/kubernetes/kubernetes/issues/120326 mentions that v5.6.0 does not include https://github.com/evanphx/json-patch/commit/50fdc0b4c9ab36933e6a0c9288fef3ac5df2b907, but it's part of v5.7.0 btw
<img width="464" alt="image" src="https://github.com/kubernetes-sigs/controller-runtime/assets/17271979/288444c6-1e16-4e4d-9e0c-a310e06e95ef">

Reply:
Maybe it's only if you use 

`github.com/evanphx/json-patch v5.6.0+incompatible`

And not for

`github.com/evanphx/json-patch/v5 v5.6.0`

Reply:
Okay so my interpretation:
* Kubernetes intentionally stayed on v4 because of: https://github.com/kubernetes/kubernetes/pull/91622#issuecomment-637087847
* For the `github.com/evanphx/json-patch` module  "v5.6.0+incompatible is older than v4.12.0+incompatible" (https://github.com/kubernetes/kubernetes/issues/120326#issue-1876406255) 
* client-go v0.29 is using `v4.12.0+incompatible` https://github.com/kubernetes/client-go/blob/v0.29.0/go.mod#L8
* we intentionally use `github.com/evanphx/json-patch` because we copied code from https://github.com/kubernetes/client-go/blob/783d0d33626e59d55d52bfd7696b775851f92107/testing/fixture.go#L146-L194

Based on all this:
* I would "upgrade" `github.com/evanphx/json-patch` to be in sync with client-go v0.29
* I would keep our regular `github.com/evanphx/json-patch/v5` dependency. I assume downgrading to v4 is a breaking change just like upgrading to v5 for Kubernetes
  * I didn't dig deeper into https://github.com/kubernetes/kubernetes/pull/91622#issuecomment-637087847 (which was the reason why Kubernetes stayed on v4)

Reply:
Ok, I've downgraded this dep to v4. I also copied the comment that you originally asked about to the go.mod, so that any changes to this line will be visible in the future. I hope this will prevent accidental changes

Reply:
Thank you!